source/common/upstream/upstream_impl.cc

```diff
@@ -240,7 +241,7 @@ HostVector filterHosts(const std::unordered_set<HostSharedPtr>& hosts,
 
 HostDescriptionImpl::HostDescriptionImpl(
     ClusterInfoConstSharedPtr cluster, const std::string& hostname,
-    Network::Address::InstanceConstSharedPtr dest_address, MetadataConstSharedPtr metadata,
+    Network::Address::InstanceConstSharedPtr dest_address, const MetadataConstSharedPtr& metadata,
```

Comment:
i think the standard guidance for passing shared ptrs is to pass by value and then move it where you want, e.g. `metadata_(std::move(metadata))`. Passing by const ref doesn't actually save you anything if you end up copying it anyways 

https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html